x-pack/plugins/rule_registry/server/alert_data_client/browser_fields/index.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FieldSpec } from '@kbn/data-plugin/common';
+import { BrowserField, BrowserFields } from '../../types';
+
+const getFieldCategory = (fieldCapability: FieldSpec) => {
+  const name = fieldCapability.name.split('.');
+
+  if (name.length === 1) {
+    return 'base';
+  }
+
+  return name[0];
+};
+
+const browserFieldFactory = (
+  fieldCapability: FieldSpec,
+  category: string
+): { [fieldName in string]: BrowserField } => {
+  return {
+    [fieldCapability.name]: {
+      ...fieldCapability,
+      category,
+    },
+  };
+};
+
+export const fieldDescriptorToBrowserFieldMapper = (
+  fieldDescriptor: FieldSpec[]
```

Comment:
nit: We can call this `fields` for this variable and when we iterate over it, use `field` for each item.